rpcs3/Emu/RSX/Common/BufferUtils.cpp

```diff
@@ -755,36 +756,286 @@ namespace
 
 	struct primitive_restart_impl
 	{
+#if __AVX2__
```

Comment:
This also blocks msvc from building with this enabled. Force-enable it in the block above the code where I also force enabled SSE3 and 4_1. This is because the solution cannot default with AVX2 since the compiler may emit AVX2 code in other places.
As for linux, well, the only portable solution here is to implement the methods as inline asm due to the SSE2 compatibility restriction, but this is not as important as for MSVC. It can be done later so that it works in appImages.

Reply:
Which SSE2 compatibility restriction?  It’s totally possible to have multiple versions of a function, with dynamic dispatch based on CPU flags, but obviously not when inlined in a single function like here.

Reply:
The rpcs3 sse2 restriction (we choose to support CPUs that old). So why do we not use march=native? I explained how other parts of the code may inadvertently generate advanced code that fails on old machines. The idea of dynamic dispatch does not actually work in practice unless you build without advanced compiler optimizations turned on.

Reply:
I added dynamic dispatch to the latest version, which I believe should work on all supported x86 CPUs.